packages/contracts-periphery/test/contracts/universal/RetroReceiver.spec.ts

```diff
@@ -0,0 +1,203 @@
+import hre from 'hardhat'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
+import { Contract } from 'ethers'
+
+import { expect } from '../../setup'
+import { deploy } from '../../helpers'
+
+describe('RetroReceiver', () => {
+  const DEFAULT_TOKEN_ID = 0
+  const DEFAULT_AMOUNT = hre.ethers.constants.WeiPerEther
+  const DEFAULT_RECIPIENT = '0x' + '11'.repeat(20)
+
+  let signer1: SignerWithAddress
+  let signer2: SignerWithAddress
+  before('signer setup', async () => {
+    ;[signer1, signer2] = await hre.ethers.getSigners()
+  })
+
+  let TestERC20: Contract
+  let TestERC721: Contract
+  let RetroReceiver: Contract
+  beforeEach('deploy contracts', async () => {
+    TestERC20 = await deploy('TestERC20', { signer: signer1 })
+    TestERC721 = await deploy('TestERC721', { signer: signer1 })
+    RetroReceiver = await deploy('RetroReceiver', {
+      signer: signer1,
+      args: [signer1.address],
+    })
+  })
+
+  beforeEach('balance setup', async () => {
+    await TestERC20.mint(signer1.address, hre.ethers.constants.MaxUint256)
+    await TestERC721.mint(signer1.address, DEFAULT_TOKEN_ID)
+    await hre.ethers.provider.send('hardhat_setBalance', [
+      DEFAULT_RECIPIENT,
+      '0x0',
+    ])
+  })
+
+  describe('constructor', () => {
+    it('should set the owner', async () => {
+      expect(await RetroReceiver.owner()).to.equal(signer1.address)
+    })
+  })
+
+  describe('receive', () => {
+    it('should be able to receive ETH', async () => {
+      await expect(
+        signer1.sendTransaction({
+          to: RetroReceiver.address,
+          value: DEFAULT_AMOUNT,
+        })
+      ).to.not.be.reverted
+    })
```

Comment:
Would be nice to have an assertion that shows the balance increasing by the expected amount